### PR TITLE
feat: Adds draft support for reply box in conversations

### LINKED ISSUE #4436: Draft messages introduces random characters in the reply box
Draft messages feature adds a random character if `\` is present in the message content. 

**To reproduce:**

- Switch to the branch which has a draft messages feature
- Type some text and add `\`, wait for a second (or click outside)
- Start typing again.

**Interim fix**: I have reverted the feature for now

### PR DESCRIPTION
**Pull request by nithindavid**
Fixes #4436 #4876

### CODE REVIEW

**Comment by nithindavid** [comment]
@pranavrajs @fayazara These are the cases I checked. Let me know if i missed something 

 https://www.loom.com/share/12d795442021437e961d8a56f7baa235

**Comment by pranavrajs** [comment]
@nithindavid Can you please resolve the conflicts?

**Review by iamsivin — DISMISSED** [review]
Code LGTM 👍🏻. Please resolve the conflict.

**Comment by pranavrajs** [comment]
@nithindavid 

1. Please resolve the conflict.
2. Create a loom video with the test cases and get it ready for review again.

**Comment by nithindavid** [comment]
@pranavrajs Please find the loom video with the cases - https://www.loom.com/share/eda07182f4fd4273ada68fd851cfb260

**Comment by nithindavid** [comment]
@pranavrajs Can you review this?

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 1]
```diff
@@ -148,6 +148,8 @@ import {
   MAXIMUM_FILE_UPLOAD_SIZE_TWILIO_SMS_CHANNEL,
 } from 'shared/constants/messages';
 import { BUS_EVENTS } from 'shared/constants/busEvents';
+import { debounce } from 'shared/helpers/TimeHelpers';
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:151]: @nithindavid Can use debounce from chatwoot/utils

**Review by iamsivin — COMMENTED** [review]
@nithindavid @pranavrajs Tested it locally Looks good, Working most of the time.
https://www.loom.com/share/da73cc99f979429eb843a8ae2d76eb20

**Issues found**

1. Type something in a conversation input field (convId- 1) and then click another conversation next to it (convId- 2) and add messages in that conversation input field and go back to another one (convId- 3) and type something and it will save and then go back to the previous saved one (convId- 2) and edit its input value and it will save, then click another conversation. Then go back to the conversation with (convId- 2) the value is different and that different value will save as draft. You can see it loom video attached below. https://www.loom.com/share/b4cfa81307e14424af31cb0d4e46abc0

2. Draft message for private note is not working well. You can see it loom video attached below. 
https://www.loom.com/share/4cd03b72500f4eac8c1c994bd5956301

3. Not rendering the rich content properly.
https://www.loom.com/share/fafc1faef3cc491b951ba4410b02b827

**Comment by iamsivin** [comment]
**I tested it locally, and it's working as expected. All the issues mentioned above were resolved. You can see it in the attached loom video.**
https://www.loom.com/share/91a9baace2aa478d8a80e8c748ad0fdb
https://www.loom.com/share/b74fda3257e7401fb337619566210f4e




> 1. Type something in a conversation input field (convId- 1) and then click another conversation next to it (convId- 2) and add messages in that conversation input field and go back to another one (convId- 3) and type something and it will save and then go back to the previous saved one (convId- 2) and edit its input value, and it will save, then click another conversation. Then go back to the conversation with (convId- 2) the value is different and that different value will save as draft. You can see it loom video attached below. 

> 2. Draft message for private note is not working well. You can see it loom video attached below.

> 3. Not rendering the rich content properly.

**Review by iamsivin — DISMISSED** [review]
I tested this PR locally and LGTM 💯.

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**